authentication/views.py

```diff
@@ -788,3 +791,64 @@ def post(self, request, **args):
             },
             status=status.HTTP_200_OK
         )
+
+
+class SavedCardsListView(generics.GenericAPIView):
+    """get all a  cards for authenticated user"""
+
+    permission_classes = (IsAuthenticated,)
+    serializer_class = CardInfoSerializer
+    def get_queryset(self):
+
+        user = self.request.user
+        return CardInfo.objects.filter(user_id=user.id)
+
+    def get(self, request):
+        serializer = self.serializer_class(self.get_queryset(), many=True)
+        if not serializer.data:
+            response = {
+                "data": {
+                    "message": "You don't have any card saved"
+                }
+            }
+            status_code = status.HTTP_404_NOT_FOUND
+        else:
+            response = {
+                "data": {
+                    "saved_cards": serializer.data,
+                    "message": "Cards retrieved",
+                }
+            }
+            status_code = status.HTTP_200_OK
+        return Response(response, status=status_code)
+
+
+class DeleteSavedCardView(generics.GenericAPIView):
+    """ Remove existing card """
+
+    permission_classes = (IsAuthenticated, IsCardOwner)
+
+    def get_object(self, request, card_info_id):
+        """
+        Handles getting a specific card
+
+        :param request:
+        :param user_id:
+        :return: Object
+        """
+        try:
+            card = CardInfo.objects.get(id=card_info_id)
+            return card
+        except CardInfo.DoesNotExist:
+            raise NotFound(detail={"errors": "Card not saved"},
+                           code=status.HTTP_404_NOT_FOUND)
+
+    def delete(self, request, **kwargs):
+        """
+        Handles removing a saved card
+        :param request:
+        :return:
+        """
+        card = self.get_object(request, self.kwargs['id'])
+        card.delete()
+        return Response("Card Deleted Successfully", status.HTTP_200_OK)
```

Comment:
Hi @Niyitangasam , why did you opt for returning plain text here as the response? what do you think of having an object like in `deleting a property`

https://github.com/landvilleng/landville-backend-web-api/blob/75eb52bb3163af99d680caf56555974e635fedcd/property/views.py#L312

Reply:
Hey @Tevinthuku, I worked on that.